sys/kern/vm_object.c

```diff
@@ -57,34 +47,39 @@ bool vm_object_add_page(vm_object_t *obj, off_t offset, vm_page_t *page) {
   page->offset = offset;
 
   SCOPED_MTX_LOCK(&obj->mtx);
+  vm_page_t *pg;
 
-  if (!RB_INSERT(vm_pagetree, &obj->tree, page)) {
-    obj->npages++;
-    vm_page_t *next = RB_NEXT(vm_pagetree, &obj->tree, page);
-    if (next)
-      TAILQ_INSERT_BEFORE(next, page, obj.list);
-    else
-      TAILQ_INSERT_TAIL(&obj->list, page, obj.list);
-    return true;
+  TAILQ_FOREACH (pg, &obj->list, obj.list) {
+    if (vm_page_cmp(page, pg) < 0) {
+      TAILQ_INSERT_BEFORE(pg, page, obj.list);
+      obj->npages++;
+      return true;
+    } else if (vm_page_cmp(pg, page) == 0) {
+      /* page at offset we are trying to add already exists in the object */
+      return false;
+    }
   }
 
-  return false;
+  /* offset of page is greater than the offset of any other page */
+  TAILQ_INSERT_TAIL(&obj->list, page, obj.list);
+  obj->npages++;
+
+  return true;
 }
 
-static void vm_object_remove_page_nolock(vm_object_t *obj, vm_page_t *page) {
```

Comment:
I think in other places we do use `_nolock` convention. Please revert that change.

Reply:
done